pkg/deploy/api/validation/validation.go

```diff
@@ -47,6 +48,53 @@ func ValidateDeploymentConfigUpdate(newConfig *deployapi.DeploymentConfig, oldCo
 	return allErrs
 }
 
+func ValidateScale(scale *extensions.Scale) fielderrors.ValidationErrorList {
+	allErrs := fielderrors.ValidationErrorList{}
+	allErrs = append(allErrs, validation.ValidateObjectMeta(&scale.ObjectMeta, true, validation.NameIsDNSSubdomain).Prefix("metadata")...)
+
+	if scale.Spec.Replicas < 0 {
+		allErrs = append(allErrs, fielderrors.NewFieldInvalid("spec.replicas", scale.Spec.Replicas, "must be non-negative"))
+	}
+
+	return allErrs
+}
+
+func ValidateScaleUpdate(newScale *extensions.Scale, oldScale *extensions.Scale) fielderrors.ValidationErrorList {
```

Comment:
I would expect this to use upstream validation and we'd have one addition check for the kind.